.github/workflows/ci.yml

```diff
@@ -0,0 +1,280 @@
+                  component/test_network_offering
+                  component/test_non_contiguous_vlan",
+                "component/test_persistent_networks
+                  component/test_project_configs
+                  component/test_project_limits
+                  component/test_project_resources",
+                "component/test_project_usage
+                  component/test_protocol_number_security_group
+                  component/test_public_ip
+                  component/test_resource_limits",
+                "component/test_regions_accounts
+                  component/test_routers
+                  component/test_snapshots
+                  component/test_stopped_vm
+                  component/test_tags
+                  component/test_templates
+                  component/test_updateResourceCount
+                  component/test_update_vm",
+                "component/test_volumes
+                  component/test_vpc
+                  component/test_vpc_distributed_routing_offering
+                  component/test_vpc_network
+                  component/test_vpc_offerings
+                  component/test_vpc_routers
+                  component/test_vpn_users",
+                "component/test_vpc_network_lbrules" ]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          architecture: x64
+          cache: maven
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+
+      - name: Install Build Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y git uuid-runtime genisoimage netcat ipmitool build-essential libgcrypt20 libgpg-error-dev libgpg-error0 libopenipmi0 ipmitool libpython3-dev libssl-dev libffi-dev python3-openssl python3-dev python3-setuptools
+
+      - name: Install Python dependencies
+        run: |
+          python3 -m pip install --user --upgrade urllib3 lxml paramiko nose texttable ipmisim pyopenssl pycrypto mock flask netaddr pylint pycodestyle six astroid
+
+      - name: Env details
+        run: |
+          uname -a
+          whoami
+          javac -version
+          mvn -v
+          python3 --version
+          free -m
+          nproc
+          git status
+          ipmitool -V
+
+      - name: Setup MySQL Server
+        run: |
+          # https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#mysql
+          sudo apt-get install -y mysql-server
+          sudo systemctl start mysql
+          sudo mysql -uroot -proot -e "ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY ''; FLUSH PRIVILEGES;"
+          sudo systemctl restart mysql
+          sudo mysql -uroot -e "SELECT VERSION();"
+
+      - name: Build with Maven
```

Comment:
This step builds on each of the matrix jobs (17x) for 10-15minutes so ~ 3 hours of runner time for something that IIUC (I am not a java person :D)  could be done in an initial job, the built package uploaded as artifacts and redistributed to the matrix jobs where it only needs to be installed. (this would also allow you to de-duplicate installation of some of the build dependencies I guess but those don't take as long) 

You can use `needs: <job_name>` to create a dependency between the build job and the matrix jobs and https://github.com/actions/upload-artifact to share the artifacts.

Reply:
with an over all runtime of 1d 10h 26m 24s any savings are important...

Reply:
Thanks @assignUser I thought about that, I'll explore. In fact because of the sheer amount of tests and duration I'm also thinking about:
- Running on tests based on type of changes (based on filesystem paths/wildcards)
- Context builds for example - not running these java builds&tests say if the changes are in UI
However, if Travis CI will be removed for asf repos by end of Dec, these optimisations can happen as a separate iteration/PR

Reply:
so you are saying move (just) these three lines to another yml, right? can you point to an example for the plumbing @assignUser ?
I think we can leave this for an improvement btw, if the job runs we should start from there (I know, we will slack, but the build times will keep reminding us)

Reply:
@DaanHoogland  I think @rohityadavcloud is on it :D but yes you would add another `job:` with those build steps + artifact upload and dependency installation and have the other jobs depend on it with `needs: build-job`

Very much +1 on context builds, sounds good!